source/configuration_manual/sql.rst

```diff
@@ -0,0 +1,527 @@
+
+``hostaddr``
+------------
+Alias for ``host``.
+
+``port``
+--------
+- Default: ``9042``
+- Values: :ref:`uint`
+
+``user``
+--------
+- Default: <empty>
+- Values: :ref:`string`
+
+Username for authentication.
+
+``password``
+------------
+- Default: <empty>
+- Values: :ref:`string`
+
+Password for authentication.
+
+``dbname``
+----------
+- Default: <empty>
+- Values: :ref:`string`
+
+Database name to connect.
+
+``client_flags``
+----------------
+- Default: 0
+- Values: :ref:`uint`
+
+Flags to use when connecting to the database, provided as base-10 number.
+
+.. seealso:: https://dev.mysql.com/doc/c-api/8.0/en/mysql-real-connect.html
+
+``connect_timeout``
+-------------------
+- Default: 5
+- Values: :ref:`uint`
+
+How many seconds to wait for connection.
+
+``read_timeout``
+----------------
+- Default: 30
+- Values: :ref:`uint`
+
+Timeout in seconds when reading data from server.
+
+``write_timeout``
+-----------------
+- Default: 30
+- Values: :ref:`uint`
+
+Timeout in seconds when writing data to server.
+
+``ssl_ca``
+----------
+- Default: <empty>
+- Values: :ref:`string`
+
+Path to SSL certificate authority file to use to validate peer certificate.
+
+``ssl_ca_path``
+---------------
+- Default: <empty>
+- Values: :ref:`string`
+
+Path to directory of SSL certificate authority files to use to validate peer certificate.
+
+``ssl_cert``
+------------
+- Default: <empty>
+- Values: :ref:`string`
+
+Path to a certificate file to use for authenticating against the remote server.
+
+``ssl_key``
+-----------
+- Default: <empty>
+- Values: :ref:`string`
+
+Path to private key matching ``ssl_cert`` to use for authenticating against the remote server.
+
+``ssl_cipher``
+---------------
+- Default: <library dependant>
+- Values: :ref:`string`
+
+Cipher to use when connecting. See client library documentation.
+
+``ssl_verify_server_cert``
+--------------------------
+- Default: no
+- Values: :ref:`boolean`
+
+Whether to verify server certificate.
+
+``option_file``
+---------------
+- Default: <empty>
+- Values: :ref:`string`
+
+File to read for client library specific configuration.
+
+``option_group``
+----------------
+- Default: <empty>
+- Values: :ref:`string`
+
+Section name to read from ``option_file``.
+
+PostgreSQL
+==========
+
+Driver name is ``postgresql``.
+
+To compile support for this driver, you need PostgreSQL client library and headers.
+
+Supported options
+-----------------
+
+PostgreSQL parses connection string directly. Doveot only parses ``host`` argument.
+
+.. seealso:: https://www.postgresql.org/docs/9.0/libpq-connect.html
+
+SQLite
+======
+
+Driver name is ``sqlite``.
+
+To compile support for this driver. you need sqlite library and headers.
+
+Supported options
+-----------------
+
+.. versionchanged:: v2.3.18
+
+Prior v2.3.18, Dovecot uses the whole value as filename to connect, whitespace included.
+This was changed to support options, so it will only use initial non-whitespace part as filename.
```

Comment:
Oh, didn't realize before, but there are no other settings in Dovecot that behave this way, better not add yet another different behavior. For example to make it work similarly as passwd-file you could just treat all parameters with "=" as options and when = isn't found that's the path. Or just that the last parameter is the path.